l1-contracts/contracts/common/libraries/openzeppelin/IncrementalMerkle.sol

```diff
@@ -122,24 +98,32 @@ library DynamicIncrementalMerkle {
             }
 
             // Compute the current node hash by using the hash function
-            // with either the its sibling (side) or the zero value for that level.
-            currentLevelHash = fnHash(
+            // with either its sibling (side) or the zero value for that level.
+            currentLevelHash = Hashes.Keccak256(
```

Comment:
Why do we need all these changes? They seem to only use `Hashes.Keccak256` instead of generalized `fnHash`, does it give performance boost?

Reply:
Function pointer was removed from the struct in upstream in this PR: https://github.com/OpenZeppelin/openzeppelin-contracts/pull/5080 

And since we're only using keccak anyway I thought we could simplify the interface of `push` a bit

Reply:
Also the changes fix a small bug with incorrect array length